payments/utils.py

```diff
@@ -0,0 +1,132 @@
+    return f"{round_price(value)}€"
+
+
+def currency(func):
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        value = func(*args, **kwargs)
+        return currency_format(value)
+
+    return wrapped
+
+
+def percentage_format(value):
+    return f"{round_price(value)}%"
+
+
+def percentage(func):
+    @wraps(func)
+    def wrapped(*args, **kwargs):
+        value = func(*args, **kwargs)
+        return percentage_format(value)
+
+    return wrapped
+
+
+def convert_aftertax_to_pretax(aftertax_price, tax_percentage):
+    return aftertax_price / (1 + tax_percentage / 100)
+
+
+def calculate_order_due_date(delta=14):
+    return date.today() + timedelta(days=delta)
+
+
+def _calculate_price_for_partial_month(
+    month_date: date, base_price: Decimal
+) -> Decimal:
+    _, days_in_month = calendar.monthrange(month_date.year, month_date.month)
+    price_per_day = base_price / days_in_month
+    return (days_in_month - month_date.day) * price_per_day
+
+
+@rounded
+def calculate_product_partial_month_price(
+    base_price: Decimal, start_date: date, end_date: date
+) -> Decimal:
+    price = Decimal(0)
+
+    # If the dates are on the same month/year
+    if start_date.month == end_date.month and start_date.year == end_date.year:
+        price = _calculate_price_for_partial_month(start_date, base_price)
+    else:
+        # Calculate price for the start month
+        price += _calculate_price_for_partial_month(start_date, base_price)
+
+        # Calculate price for the months in between
+        months_between = list(
+            rrule(
+                MONTHLY,
+                dtstart=start_date + relativedelta(day=1, months=1),
+                until=end_date - relativedelta(day=1, months=1),
+            )
+        )
+        # Calculate the added price for all the complete months in between
+        price += len(months_between) * base_price
+
+        # Calculate price for the end month
+        price += _calculate_price_for_partial_month(end_date, base_price)
+
+    return price
+
+
+@rounded
+def calculate_product_partial_season_price(
+    base_price: Decimal, start_date: date, end_date: date, summer_season: bool = True
+) -> Decimal:
+    # If it's the "normal" (summer) season, calculate with the normal dates
+    season_days = calculate_berth_lease_end_date() - calculate_berth_lease_start_date()
+    # If it's for the opposite season ("winter season"), calculate the inverse
+    if not summer_season:
+        # Calculate the amount of days in the year, in case it's a leap year
+        days_in_year = (start_date + relativedelta(years=1)) - start_date
+        season_days = days_in_year - season_days
+    delta = (end_date - start_date).days
+    price = (delta * base_price) / season_days.days
+    return price
+
+
+@rounded
+def calculate_product_partial_year_price(
+    base_price: Decimal, start_date: date, end_date: date
+) -> Decimal:
+    days_in_year = 366 if calendar.isleap(start_date.year) else 365
```

Comment:
what if the start date is 30th of September and end date is 30th of September next year AND start year is not a leap year, but end year is 🤔

Reply:
Even worse: what if:
```
start = date(2019, 9, 30)
end = date(2021, 9, 30)
```
(would that be even possible?)
Given that 2020 _is_ leap year, how should we determine the yearly price? Would it be an average of days (`(365 + 366 + 365)/3`) or do we take the price for each year individually and then add it or..?

If start/end dates weren't allowed to be over a year apart (i.e. delta < 366), we can just check for something like this?
```python
if (
    is_leap(start) or
    end.month >= MARCH and is_leap(end):
):
    days = 366
```

Reply:
yeah, I was thinking about this too yesterday. we _should not_ have lease longer than a year business-wise. they are however possible technically. I like the idea of not allowing creating orders for lease longer than 366 days 🤔

Reply:
So we can restrict them from the `leases` app..?

Reply:
hm, good idea, maybe we should do it in the leases. let's disallow for now any lease longer than 1 year

Reply:
Ok ok, then let's assume that leases won't be over a year and work on that later. 👍

Reply:
sure, let's do that 👍